rdmd.d

```diff
@@ -916,3 +898,78 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+struct FileAttributes
+{
+    uint attributes;
+    bool exists() const { return attributes != 0xFFFFFFFF; }
+    bool isFile() const
+        in { assert(exists); } do
+    {
+        import std.file : attrIsFile;
+        return attrIsFile(attributes);
+    }
+}
+
+struct Yap
```

Comment:
`yap` wasn't a great name start with, and this is not making it better. This is a namespace for wrapping filesystem operations, so let's just name it `Filesystem` or such.

Reply:
Please also add a comment describing the motivation of having this structure.

Reply:
Added a comment in #299.  I like the name `Yap` for the structure, let's come back to this if you still feel the same after this set of changes.